embedded-graphics/src/primitives/mod.rs

```diff
@@ -387,6 +495,111 @@ macro_rules! egrectangle {
     }};
 }
 
+/// Create an [`Sector`](./primitives/sector/struct.Sector.html) with optional styling using a
+/// convenient macro.
+///
+/// ```rust
+/// use embedded_graphics::{
+///     egsector,
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     primitive_style,
+///     primitives::Sector,
+///     style::{PrimitiveStyle, Styled},
+/// };
+///
+/// // Coordinates can be defined as any type that implements `Into<Point>`
+/// // Angles can be defined as any type that implements `Into<Angle>`
+/// let line_sector: Styled<Sector, PrimitiveStyle<Rgb565>> =
+///     egsector!(top_left = (10, 20), diameter = 30, angle_start = 90.0, angle_sweep = 60.0);
+///
+/// let filled_sector: Styled<Sector, PrimitiveStyle<Rgb565>> = egsector!(
+///     center = (10, 20),
+///     diameter = 30,
+///     angle_start = 90.0,
+///     angle_sweep = 60.0,
+///     style = primitive_style!(stroke_color = Rgb565::RED, fill_color = Rgb565::GREEN)
+/// );
+/// ```
+///
+/// Style properties like `stroke_color` map to methods on the [`PrimitiveStyleBuilder`] struct.
+/// For example, the following code makes two identical sectors:
+///
+/// [`PrimitiveStyleBuilder`]: style/struct.PrimitiveStyleBuilder.html
+///
+/// ```rust
+/// use embedded_graphics::{
+///     egsector,
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     primitive_style,
+///     primitives::Sector,
+///     style::{PrimitiveStyle, PrimitiveStyleBuilder, Styled},
+/// };
+///
+/// let sector_1: Styled<Sector, PrimitiveStyle<Rgb565>> = egsector!(
+///     top_left = (10, 20),
+///     diameter = 30,
+///     angle_start = 90.0,
+///     angle_sweep = 60.0,
+///     style = primitive_style!(
+///         stroke_color = Rgb565::RED,
+///         fill_color = Rgb565::GREEN,
+///         stroke_width = 1
+///     )
+/// );
+///
+/// let style = PrimitiveStyleBuilder::new()
+///     .fill_color(Rgb565::GREEN)
+///     .stroke_color(Rgb565::RED)
+///     .stroke_width(1)
+///     .build();
+///
+/// let sector_2: Styled<Sector, PrimitiveStyle<Rgb565>> =
+///     Sector::new(Point::new(10, 20), 30, Angle::from_degree(90.0), Angle::from_degree(60.0)).into_styled(style);
+///
+/// assert_eq!(sector_1, sector_2);
+/// ```
+#[macro_export]
+macro_rules! egsector {
```

Comment:
The note about angles for `egarc` also applies to this macro.

Reply:
Done.